provider/vsphere/instance.go

```diff
@@ -53,11 +53,7 @@ func (inst *environInstance) Addresses() ([]network.Address, error) {
 	res := make([]network.Address, 0)
 	for _, net := range inst.base.Guest.Net {
 		for _, ip := range net.IpAddress {
-			scope := network.ScopeCloudLocal
-			if net.Network == inst.env.ecfg.externalNetwork() {
```

Comment:
Okay, from discussing I agree trying to keep an override to public based on this config is mostly just harmful.